### ISSUE TITLE
`useNumberField` does not allow input from MacOS Japanese IME keyboard

### ISSUE THREAD

**Issue by tuananhlai** [adobe/react-spectrum#6678]
### Provide a general summary of the issue here

React Aria's NumberField does not allow input from Japanese keyboard on MacOS.

https://github.com/adobe/react-spectrum/assets/42168740/4f3782ff-298e-4ed0-97b9-39eae176768c

`<input type='number' />`

https://github.com/adobe/react-spectrum/assets/42168740/cebaf076-ce45-4e0b-9375-8ba51fea361a

### 🤔 Expected Behavior?

The number field receives value from the keyboard.

### 😯 Current Behavior

The number field does not receive value from the keyboard.

### 🖥️ Steps to Reproduce

1. Switch input to Japanese.
2. Visit https://react-spectrum.adobe.com/react-aria/useNumberField.html#example
3. Press a number on the keyboard.

### Version

latest

### What browsers are you seeing the problem on?

Firefox

### What operating system are you using?

MacOS Sonoma 14.5

**Comment by snowystinger** [comment]
This may be due to me not knowing Japanese or maybe not knowing enough about the keyboard and IME input, however, if I change to a Japenese keyboard on my mac and type the `1` key, this is what I get `ぬ` which I don't think is a number as I think that is `一`. I can't get the IME to show up on a native `input type=number` outside our docs. Could you include some more instructions for bringing it up?

That said, I have some guesses. I think we're preventing the IME from appearing possibly by not accepting any characters other than the supported numbering system symbols. Maybe we can lift that restriction during composition.

**Comment by tuananhlai** [comment]
Gladly. The keyboard I'm using is `Japanese - Romaji`. I've included a screen capture for adding it below.


https://github.com/adobe/react-spectrum/assets/42168740/0a3b4ab3-aadf-432f-899b-fe00839cacf8

I'm using `Options + Space` to change my keyboard input to Japanese, but I'm not sure if yours would be the same.

**Comment by yihuiliao** [comment]
Yeah, I was able to reproduce. It seems like it kinda works in Safari though (at least for me), and I noticed that it doesn't work in RSP NumberField either.

**Comment by tuananhlai** [comment]
The input kinda works for Safari but there's a weird bug when you are cycling between the options.


https://github.com/adobe/react-spectrum/assets/42168740/57493596-cf1d-4e5c-b638-2798a17c3c67

**Comment by wrengr** [comment]
> This may be due to me not knowing Japanese or maybe not knowing enough about the keyboard and IME input, however, if I change to a Japenese keyboard on my mac and type the `1` key, this is what I get `ぬ` which I don't think is a number as I think that is `一`. I can't get the IME to show up on a native `input type=number` outside our docs. Could you include some more instructions for bringing it up?

There are two very different things that "Japanese input/IME" can mean, and I think that's part of the confusion here (for the quote above, not necessarily for this issue more generally).

- The first sort is where you're using a western-style keyboard (ANSI, ISO, Dvorak, etc) and type things out via roomaji — that is, to type "か" you'd hit the `k` key followed by the `a` key. When using this input method, the `1` key maps to the "1" character.
- The second sort is where you're using a Japanese-style keyboard (OADG, Apple-JIS, CBB, etc) and type things out via kana — that is,  to type "か" you'd hit the `か` key. When using this input method, the `1` key (or rather, the key in that same position) maps to the "ぬ" character (it also maps to the "1" character when in the 英数 mode).

For macOS, technically you can set things up to use both IMEs regardless of the actual physical keyboard. Though using the JIS/kana IME with a western physical-keyboard will be painful, since western keyboards are missing the 5~6 physical keys for switching between different modes (Apple-JIS keyboards actually only have two of those keys, unlike OADG/Win-JIS keyboards; see https://en.wikipedia.org/wiki/Language_input_keys for more gritty details)
